### ISSUE TITLE
Embrace Extend Extinguish Protection for Invented Programming Languages

### ISSUE THREAD

**Issue by sithys** [polyformproject/polyform-licenses#82]
I am not a lawyer, and I have questions about source available licenses in the domain of recently-invented programming languages. Even tiny code changes in the implementation language can completely change the syntax and semantics of the invented language. The term "Embrace Extend Extinguish" (a.k.a. Embrace Extend Exterminate) refers to a policy used by Microsoft in response to the Java programming language. Microsoft intended to make an incompatible version of Java and then make those incompatibilities required for their own platform.

Is there a license that allows commercial use of a new programming language, but which bans making incompatible versions of the language? This ban should extend to both commercial and non-commercial entities. Any advice would be helpful.

**Comment by castarco** [comment]
This is an interesting idea for a license, although (disclaimer: I'm not a lawyer) I suspect that this constraint would be difficult to encode in a license without making it very dense.

**Comment by sithys** [comment]
Thanks for the response. I understand the difficulty of making a license that appeals to programming language inventors. In the last few years I released my language under MIT license. I also renamed my language to Moirai, though I do not have any users, because the original language name ShardScript is unfortunately a slight street slang/negative connotation. See: https://github.com/moirai-lang/moirai-kt
